app/src/main/java/de/westnordost/streetcomplete/quests/bikeway/AddCycleway.java

```diff
@@ -0,0 +1,162 @@
+			applyCyclewayAnswerTo(cyclewayRight, Side.RIGHT, cyclewayRightDir, changes);
+		}
+
+		if(answer.getBoolean(AddCyclewayForm.IS_ONEWAY_NOT_FOR_CYCLISTS))
+		{
+			changes.addOrModify("oneway:bicycle", "no");
+		}
+	}
+
+	private enum Side
+	{
+		LEFT("left"), RIGHT("right"), BOTH("both");
+
+		public final String value;
+		Side(String value) { this.value = value; }
+	}
+
+	private void applyCyclewayAnswerTo(Cycleway cycleway, Side side, int dir, StringMapChangesBuilder changes)
+	{
+		String directionValue = null;
+		if(dir != 0) directionValue = dir > 0 ? "yes" : "-1";
+
+		String cyclewayKey = "cycleway:" + side.value;
+		switch (cycleway)
+		{
+			case NONE:
+				changes.add(cyclewayKey, "no");
+				break;
+			case LANE:
+				changes.add(cyclewayKey, "lane");
+				if(directionValue != null)
+				{
+					changes.addOrModify(cyclewayKey + ":oneway", directionValue);
+				}
+				break;
+			case TRACK:
+				changes.add(cyclewayKey, "track");
+				if(directionValue != null)
+				{
+					changes.addOrModify(cyclewayKey + ":oneway", directionValue);
+				}
+				break;
+			case TRACK_DUAL:
+				changes.add(cyclewayKey, "track");
+				changes.addOrModify(cyclewayKey + ":oneway", "no");
+				break;
+			case LANE_DUAL:
+				changes.add(cyclewayKey, "lane");
+				changes.addOrModify(cyclewayKey + ":oneway", "no");
+				break;
+			case SIDEWALK:
+				// https://wiki.openstreetmap.org/wiki/File:Z240GemeinsamerGehundRadweg.jpeg
+				changes.add(cyclewayKey, "track");
+				changes.add(cyclewayKey + ":segregated", "no");
+				changes.addOrModify("sidewalk", side.value);
+				break;
+			case SIDEWALK_OK:
+				// https://wiki.openstreetmap.org/wiki/File:Z239Z1022-10GehwegRadfahrerFrei.jpeg
+				changes.add(cyclewayKey, "no");
+				changes.addOrModify("sidewalk", side.value);
+				changes.add("sidewalk:" + side.value + ":bicycle", "yes");
+				break;
+			case SHARED:
+				changes.add(cyclewayKey, "shared_lane");
+				break;
+			case BUSWAY:
+				changes.add(cyclewayKey, "share_busway");
+				break;
+		}
+	}
+
+	@Override public boolean appliesTo(Element element)
+	{
+		/* Whether this element applies to this quest cannot be determined by looking at that
+		   element alone (see download()), an Overpass query would need to be made to find this out.
+		   This is too heavy-weight for this method so it always returns false. */
+
+		/* The implications of this are that this quest will never be created directly
+		   as consequence of solving another quest and also after reverting an input,
+		   the quest will not immediately pop up again. Instead, they are downloaded well after an
+		   element became fit for this quest. */
+		return false;
+	}
+
+	@Override public boolean download(BoundingBox bbox, MapDataWithGeometryHandler handler)
+	{
+		return overpassServer.getAndHandleQuota(getOverpassQuery(bbox), handler);
+	}
+
+	/** @return overpass query string to get streets without cycleway info not near paths roads that don't have names */
```

Comment:
> overpass query string to get streets without cycleway info not near paths roads that don't have names

is quite confusing, especially "roads that don't have names" given that there is no filtering by name that I see.